activerecord/test/cases/adapters/mysql2/table_options_test.rb

```diff
@@ -41,6 +41,20 @@ def teardown
     options = %r{create_table "mysql_table_options", options: "(?<options>.*)"}.match(output)[:options]
     assert_match %r{COLLATE=utf8mb4_bin}, options
   end
+
+  test "schema dump works with NO_TABLE_OPTIONS sql mode" do
+    skip "As of MySQL 5.7.22, NO_TABLE_OPTIONS is deprecated. It will be removed in a future version of MySQL." if @connection.database_version >= "5.7.22"
+
+    old_sql_mode = @connection.exec_query("SELECT @@session.sql_mode").first["@@session.sql_mode"]
+    new_sql_mode = old_sql_mode.split(",") + ["NO_TABLE_OPTIONS"]
+    @connection.execute("SET @@session.sql_mode=\"#{new_sql_mode.join(",")}\"")
+
+    @connection.create_table "mysql_table_options", force: true
+    output  = dump_table_schema("mysql_table_options")
+    assert_no_match %r{options:}, output
+
+    @connection.execute("SET @@session.sql_mode=\"#{old_sql_mode}\"")
```

Comment:
:thinking: It is best practice to rollback those changes in `ensure` block to not pollute testing environment if test fails in the middle of execution and this rollback is not reached.

What about something similar to?

```ruby
    old_sql_mode = @connection.exec_query("SELECT @@session.sql_mode").first["@@session.sql_mode"]
    new_sql_mode = old_sql_mode.split(",") + ["NO_TABLE_OPTIONS"]

    begin
      @connection.execute("SET @@session.sql_mode=\"#{new_sql_mode.join(",")}\"")

      @connection.create_table "mysql_table_options", force: true
      output  = dump_table_schema("mysql_table_options")
      assert_no_match %r{options:}, output
    rescue
      @connection.execute("SET @@session.sql_mode=\"#{old_sql_mode}\"")
    end
```

Reply:
Yes, good idea, should be `ensure` though. I modified the commit.